src/generic/MPoly.jl

```diff
@@ -2409,11 +2427,15 @@ function ^(a::MPoly{T}, b::Int) where {T <: RingElement}
       end
       return parent(a)([coeff(a, 1)^b], exps)
    elseif b == 0
-      return parent(a)(1)
+      return one(parent(a))
    elseif b == 1
-      return a
+      return deepcopy(a)
    elseif b == 2
       return a*a
+   elseif !hasmethod(characteristic, Tuple{T}) ||
```

Comment:
Just an idea: what about defining a fallback `charasteristic(x) = missing` ? Then the `hasmethod` call can be avoided (it's not very efficient, even though in this case it probably doesn't matter).

Reply:
Yes, I will do this.

Reply:
Done.